dockertest/dockercmd.py

```diff
@@ -398,15 +398,21 @@ def wait(self, timeout=None):
     @property
     def done(self):
         """
-        Return True if processes has ended
+        Return True if processes has ended or timeout reached
 
         :raises DockerTestError: on incorrect usage
         """
 
         if self._async_job is None:
             raise DockerTestError("Attempted to wait for done before execute()"
                                   " called.")
-        return self._async_job.sp.poll() is not None
+        has_exited = self._async_job.sp.poll() is not None
+        timed_out = self.duration >= self.timeout
+        # Don't assume this condition is logged elsewhere
+        if timed_out:
+            self.subtest.logwarning("Async. docker command has timed out"
```

Comment:
...though.  I wonder if not raising an exception could cause a false-positive situation...hmmmm, maybe you're right.